doc/inband-telemetry/SAI-Proposal-Advanced-Telemetry-draft.md

```diff
@@ -0,0 +1,740 @@
+
+# Overview
+
+This draft describes the API proposal to enable advanced telemetry capabilities in a network device. The main goal of this advanced telemetry proposal is to achieve per-packet network visibility with low overhead. The network device should be able to inspect and take telemetry actions on each individual data packet.
+
+
+## Telemetry data plane workflow
+
+![workflow](figures/workflow.png "Figure 1: Telemetry data plane workflow")
+
+__Figure 1: Telemetry data plane workflow__
+
+Figure 1 describes the telemetry data plane functionality at a high level. The telemetry module inspects every data packet without interfering with normal data forwarding pipeline. Different components may be located at different switches in the network. A telemetry watchlist table specifies the flows to monitor. It performs ternary match on the packet headers and switch ports, and provides telemetry action parameters. Packets belonging to the specified flow spaces will be processed by the event detection logic. If a triggering event is detected, the switch will generate a report message to the monitor. The report message includes packet header and switch metadata associated with the packet (e.g., timestamp, ingress/egress ports, queue depth/latency).
+
+Sections below introduce four different data plane telemetry capabilities: In-band Network Telemetry (INT), Packet Postcards, and Mirror on Drop, and Queue Congestion Report.
+
+## In-band Network Telemetry
+
+![INT](figures/INT.png "Figure 2: In-band Network Telemetry")
+
+__Figure 2: In-band Network Telemetry__
+
+Figure 2 shows an example workflow of In-band Network Telemetry (INT). Switches along the route path add switch metadata into the packet header based on the *telemetry instructions* carried in the INT header.
+
+Each switch may play the role of __endpoint__ or __transit__ for INT packets. INT endpoint acts both as __INT source__ and __INT sink__. As INT source it initiates INT operation by inserting INT header into a packet and thereby instructing other network devices along the routing path to add desired telemetry information into the packet. As INT sink it extracts the INT information from the packets and send telemetry reports to the monitor.
+
+An __INT transit__ is a device along the path of a packet from source to sink. INT transit device adds its own INT information to the packet as requested by an INT source.
+
+
+## Packet Postcards
+
+![Postcard](figures/postcard.png "Figure 3: Postcard")
+
+__Figure 3: Packet Postcards__
+
+Figure 3 shows an example workflow of packet Postcards. Each switch makes its own decision and reports packets info (aka “postcards”) to the monitor individually. Unlike INT, a Postcard switch never modifies the original data packets. INT and Postcard achieve the same flow monitoring capabilities with different data plane approaches. They are __mutually exclusive__ in a network deployment.
```

Comment:
post card and int should not be mutually exclusive.

Reply:
Removed the sentence